source/backend/api/Services/CompensationRequisitionService.cs

```diff
@@ -142,7 +151,131 @@ private static string GetCompensationRequisitionStatusText(bool? isDraft)
+        private static DateOnly? CheckFinalizedDate(bool? currentStatusIsDraft, bool? newStatusIsDraft, DateOnly? currentValue)
+        {
+            if (currentStatusIsDraft.Equals(newStatusIsDraft))
+            {
+                return currentValue;
+            }
+
+            if (newStatusIsDraft.HasValue)
+            {
+                return newStatusIsDraft.Value ? null : DateOnly.FromDateTime(DateTime.UtcNow);
+            }
+
+            return null;
+        }
+
+        private PimsCompensationRequisition AddAcquisitionFileCompReq(PimsCompensationRequisition compensationRequisition)
+        {
+            _user.ThrowIfNotAuthorized(Permissions.CompensationRequisitionAdd);
+
+            compensationRequisition.ThrowIfNull(nameof(compensationRequisition));
+            if (compensationRequisition.AcquisitionFileId is null)
+            {
+                throw new BadRequestException("Invalid acquisitionFileId.");
+            }
+
+            _logger.LogInformation("Adding compensation requisition for acquisition file id: {acquisitionFileId}", compensationRequisition.AcquisitionFileId);
+
+            if (compensationRequisition.LeaseId is not null)
+            {
+                throw new BadRequestException("Compensation Requisition should have only one parent Id");
+            }
+
+            _user.ThrowInvalidAccessToAcquisitionFile(_userRepository, _acqFileRepository, (long)compensationRequisition.AcquisitionFileId);
+
+            _ = _acqFileRepository.GetById((long)compensationRequisition.AcquisitionFileId) ?? throw new BadRequestException("Invalid acquisitionFileId.");
+
+            compensationRequisition.IsDraft ??= true;
+            var newCompensationRequisition = _compensationRequisitionRepository.Add(compensationRequisition);
+            _compensationRequisitionRepository.CommitTransaction();
+
+            return newCompensationRequisition;
+        }
+
+        private PimsCompensationRequisition AddLeaseFileCompReq(PimsCompensationRequisition compensationRequisition)
+        {
+            _user.ThrowIfNotAuthorized(Permissions.CompensationRequisitionAdd);
+
+            compensationRequisition.ThrowIfNull(nameof(compensationRequisition));
+            if (compensationRequisition.LeaseId is null)
+            {
+                throw new BadRequestException("Invalid LeaseId.");
+            }
+
+            _logger.LogInformation("Adding compensation requisition for lease file id: {leaseId}", compensationRequisition.LeaseId);
+
+            if (compensationRequisition.AcquisitionFileId is not null)
+            {
+                throw new BadRequestException("Compensation Requisition should have only one parent Id");
+            }
+
+            var pimsUser = _userRepository.GetByKeycloakUserId(_user.GetUserKey());
+            var currentLease = _leaseRepository.GetNoTracking((long)compensationRequisition.LeaseId) ?? throw new BadRequestException("Invalid LeaseId.");
```

Comment:
this already throws an exception if the lease id isn't found, so the request exception is redundant.

Reply:
removed